class-fp-fetch-remote-posts.php

```diff
@@ -0,0 +1,388 @@
+<?php
+/**
+ * This file contains the class that imports posts from remote WordPress URL.
+ *
+ * @link     https://github.com/vermadarsh/fetch-posts-rest-api-wpcli
+ * @package  FP_Fetch_Remote_Posts
+ * @since    1.0.0
+ */
+
+/**
+ * Exit if accessed directly.
+ */
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ *
+ * Check if the custom class exists.
+ */
+if ( ! class_exists( 'FP_Fetch_Remote_Posts' ) ) :
+	/**
+	 * Main class.
+	 *
+	 * This is used to define functions that import posts from remote WordPress URL.
+	 *
+	 * @since      1.0.0
+	 * @package    FP_Fetch_Remote_Posts
+	 * @author     Adarsh Verma <adarsh.verma@rtcamp.com>
+	 */
+	class FP_Fetch_Remote_Posts {
+
+		/**
+		 * This function accepts argument, sets the WP CLI command to receive url argument.
+		 *
+		 * @param array $args Accepts the list of indexed arguments.
+		 * @param array $assoc_args Accepts the list of associative arguments.
+		 */
+		public function from( $args, $assoc_args ) {
+
+			$url = WP_CLI\Utils\get_flag_value( $assoc_args, 'website' );
+			if ( ! empty( $url ) && filter_var( $url, FILTER_VALIDATE_URL ) ) :
+
+				$post_types = WP_CLI\Utils\get_flag_value( $assoc_args, 'post_type' );
+				if ( ! empty( $assoc_args['post_type'] ) ) :
+
+					/**
+					 * Check if this argument sets for multiple CPTs.
+					 */
+					if ( false !== stripos( $post_types, ',', true ) ) :
+
+						$post_types = explode( ',', $post_types );
+						if ( ! empty( $post_types ) && is_array( $post_types ) ) :
+
+							foreach ( $post_types as $post_type ) :
+								$this->frp_import_posts( $url, $post_type );
+							endforeach;
+
+						endif;
+
+					else :
+						$this->frp_import_posts( $url, $post_types );
+					endif;
+
+				else :
+					$this->frp_import_posts( $url, 'posts' );
+				endif;
```

Comment:
You can leverage default value param of `get_flag_value` to reduce redundancy.

```
						$post_types = WP_CLI\Utils\get_flag_value( $assoc_args, 'post_type', 'post' );
						
						$post_types = explode( ',', $post_types );
						if ( ! empty( $post_types ) && is_array( $post_types ) ) :

							foreach ( $post_types as $post_type ) :
								$this->frp_import_posts( $url, $post_type );
							endforeach;

						endif;
```

I believe this would do the same. `explode` will return a whole string in an array element if it doesn't find `,` delimiter. So you can use the above snippet to reduce redundancy. Please refer https://www.php.net/manual/en/function.explode.php